### ISSUE TITLE
Type checking error in SSP DD: `f_i`

### ISSUE THREAD

**Issue by balacij** [JacquesCarette/Drasil#3123]
Related to #3093 

```
`f_i` exposes ill-typed expressions!
  - ERROR: Expressions in case statement contain different types.
```

https://github.com/JacquesCarette/Drasil/blob/0a240c0231b6f737011bc090962ae1925fbc83e2/code/drasil-example/ssp/lib/Drasil/SSP/DataDefs.hs#L182-L196

**Comment by JacquesCarette** [comment]
See #3122 for the analysis of what's also going on here (vector problems).

**Comment by JacquesCarette** [comment]
And same as #3122: this symptom is fixed, and the new on is in #3124.
